### PR TITLE
First draft for replacing the legacy search with GraphQL

### PR DESCRIPTION
**Pull request by royderks**

### CODE REVIEW

File: src/app/utils/usePagination.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { useQuery } from 'urql'
+
+const usePreviousValue = track => {
```
Comment (chrisvanmook) [inline src/app/utils/usePagination.js]: could you rename the method and parameter to sth more verbose?

File: src/app/utils/usePagination.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const ref = React.useRef(track)
+  React.useEffect(() => {
+    ref.current = track
+  })
+  return ref.current
+}
+
+const usePagination = (query, input, limit = 50, initialFrom) => {
```
Comment (chrisvanmook) [inline src/app/utils/usePagination.js]: can you rename initialFrom and give it a default value (or switch limit and initialFrom)?

File: src/app/utils/usePagination.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const [result, setResult] = React.useState({})
+  const [from, setFrom] = React.useState(initialFrom)
```
Comment (chrisvanmook) [inline src/app/utils/usePagination.js]: what is from?

Comment (royderks) [reply src/app/utils/usePagination.js]: It's the offset, so you have `from` and `limit` like this `from=0&limit=50` = first fifty

**Pull request merged**